Octokit/Models/Request/NewRepositoryWebHook.cs

```diff
@@ -0,0 +1,132 @@
+    /// </para>
+    /// </remarks>   
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class NewRepositoryWebHook : NewRepositoryHook
+    {
+        /// <summary>
+        /// Initializes a new instance of the<see cref="NewRepositoryWebHook"/> class.   
+        /// Using default values for ContentType, Secret and InsecureSsl.     
+        /// </summary>
+        /// <param name="name">Use web for a webhook or use the name of a valid service. (See /hooks for the list of valid service names.)
+        /// Use "web" for a webhook or use the name of a valid service. (See 
+        /// <see href="https://api.github.com/hooks">https://api.github.com/hooks</see> for the list of valid service
+        /// names.)
+        /// </param>
+        /// <param name="config">
+        /// Key/value pairs to provide settings for this hook. These settings vary between the services and are
+        /// defined in the github-services repository. Booleans are stored internally as “1” for true, and “0” for
+        /// false. Any JSON true/false values will be converted automatically.
+        /// </param>
+        /// <param name="url">
+        /// A required string defining the URL to which the payloads will be delivered.
+        /// </param>
+        public NewRepositoryWebHook(string name, IReadOnlyDictionary<string, string> config, string url) 
+            : base(name, config)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(url, "url");
+
+            Url = url;
+            ContentType = WebHookContentType.Form;
+            Secret = "";
+            InsecureSsl = false;
+        }
+
+        /// <summary>
+        /// Gets the URL of the hook to create.
+        /// </summary>
+        /// <value>
+        /// The URL.
+        /// </value>
+        public string Url { get; protected set; }
+
+        /// <summary>
+        /// Gets the content type used to serialize the payload. The default is `form`.
+        /// </summary>
+        /// <value>
+        /// The content type.
+        /// </value>
+        public WebHookContentType ContentType { get; set; }
+        
+        /// <summary>
+        /// Gets the secret used as the key for the HMAC hex digest
+        /// of the body passed with the HTTP requests as an X-Hub-Signature header.
+        /// </summary>
+        /// <value>
+        /// The secret.
+        /// </value>
+        public string Secret { get; set; }
+        
+        /// <summary>
+        /// Gets wether the SSL certificate of the host will be verified when 
+        /// delivering payloads. The default is `false`.
+        /// </summary>
+        /// <value>
+        ///  <c>true</c> if SSL certificate verification is not performed; 
+        /// otherwise, <c>false</c>.
+        /// </value>
+        public bool InsecureSsl { get; set; }
+
+        public override NewRepositoryHook ToRequest()
+        {
+            var config = Config.Union(new Dictionary<string, string>
```

Comment:
So right now, it will fail with the following error `An item with the same key has already been added.` :bomb: This is of course due to the way the equality comparison on a key value pair checks the object, not just the key.

So we can introduce a custom equality comparer

```csharp
public class WebHookConfigComparer : IEqualityComparer<KeyValuePair<string, string>>
    {
        public bool Equals(KeyValuePair<string, string> x, KeyValuePair<string, string> y)
        {
            return x.Key == y.Key;
        }

        public int GetHashCode(KeyValuePair<string, string> obj)
        {
            return obj.Key.GetHashCode();
        }
    }
```

and use that like this

```csharp
var allConfigs = Config.Union(new Dictionary<string, string>
            {
                { "url", Url },
                { "content_type", ContentType.ToParameter() },
                { "secret", Secret },
                { "insecure_ssl", InsecureSsl.ToString() }
            }, new WebHookConfigComparer());
```

This way we will only include distinct key/value pairs based on the key itself, the value that would be used is the one that was passed in the `config` parameter in the `ctor`. Personally I'm leaning to that we should be using the values of the `NewRepositoryWebHook` properties to override the values in the `config` parameter...although, maybe we should throw a helpful error message if they pass web hook specific config values in the `config` parameter...thoughts?